_api-reference/ingest-apis/geoip.md

```diff
@@ -0,0 +1,172 @@
+
+#### Example: JSON response
+
+```json
+{
+    "acknowledged":true
+}
+```
+
+## Sending a GET request
+
+To request data from the specifed database you created, send a GET request.  
+
+#### Example: GET request
+
+```
+GET https://<host>:<port>/_plugins/geospatial/ip2geo/datasource/_all
+```
+
+#### Example: GET response
+
+```
+<insert-response-example>
+```
+
+The reponse shows information for each field (for example, `name`, `endpoint`, `provider`) in the data source file, when the data source file last updated successfully or failed (for example, `last_succeeded_at_in_epoch_millis`), and fields (for example, `fields`) added to the file since you last updated it.
+
+## Updating the data source
+
+To update the data source file, send a PUT request. You can continue using the current endpoint value or change it. Note that if the new endpoint value contains fields that are not in the current data source file, the update fails. You also can change the update interval.  
+
+#### Example: PUT request
+
+```json
+{
+    "endpoint": https://geoip.maps.opensearch.org/v1/geolite2-city/manifest.json
+    "update_interval_in_days": 1
+}
+```
+
+#### Example: Response
+
+```json
+{
+    "acknowledged":true
+}
+```
+
+## Creating the IP2geo processor
+
+Once the data source is created, you can create the `Ip2geo` processor. To create the processor, send a PUT request.
+
+#### Example: PUT request
+
+```json
+{
+   "description":"convert ip to geo",
+   "processors":[
+    {
+        "ip2geo":{
+            "field":"_ip",
+            "datsource"::"test1"
+        }
+    }
+   ] 
+}
+
+
+#### Example: Response
+
+```json
+<insert-example-response>
+```
+
+## Using the IP2geo processor in a pipeline
+
+The following table describes <what>.
+
+| Name | Required | Default | Description |
+|------|----------|---------|-------------|
+| field | yes | - | The field to get the ip address from for the geographical lookup. |
+| target_field | no | ip2geo | The field that will hold the geographical information looked up from the Maxmind database. |
+| database | no | geoip.maps.opensearch.org/v1/geolite2-city/manifest.json | The database filename referring to a database the module ships with or a custom database in the ingest-geoip config directory. |
+| properties | no |  Default field depends on what is available in `database`. | Controls what properties are added to the target_field based on the geoip lookup. |
+| ignore_missing | no | false | If `true` and `field` does not exist, the processor quietly exits without modifying the document. |
+| first_only | no | true | If `true` only first found geoip data will be returned, even if field contains array. |
+
+
+The following code is an example of using <example_name> and adds the geographical information to the `geoip` field based on the `ip` field.
```

Comment:
The following code is an example of using IP2geo processor to adds the geographical information to the `ip2geo` field based on `ip` field.

```
PUT /_ingest/pipeline/ip2geo
{
   "description":"convert ip to geo",
   "processors":[
    {
        "ip2geo":{
            "field":"ip",
            "datasource":"my-datasource"
        }
    }
   ] 
}

PUT /my-index/_doc/my-id?pipeline=ip2geo
{
  "ip": "172.0.0.1"
}

GET /my-index/_doc/my-id
```
Which returns:
```
{
   "_index":"my-index",
   "_id":"my-id",
   "_version":1,
   "_seq_no":0,
   "_primary_term":1,
   "found":true,
   "_source":{
      "my_ip_field":"172.0.0.1",
      "ip2geo":{
         "continent_name":"North America",
         "region_iso_code":"AL",
         "city_name":"Calera",
         "country_iso_code":"US",
         "country_name":"United States",
         "region_name":"Alabama",
         "location":"33.1063,-86.7583",
         "time_zone":"America/Chicago"
      }
   }
}
```